### DISCUSSION TITLE
Tapping on grabber handle

### DISCUSSION THREAD

**Discussion by shuna** [scenee/FloatingPanel#478] [Q&A]
Hi, thank you for this brilliant library. I'm working on embedding FloatingPanel to my app.
Is there any possibility to detect tap on Glabber handle and move each floating state cyclically with each tap?

I tried to add TapGestureRecognizer to FloatingPanelController.surfaceView.grabberHandle but it does not work. Also I added TapGestureRecognizer to FloatingPanelController.surfaceView but it seem to interfere with tapping its content (UICollectionViewCell selection).

Thanks.

**Comment by scenee** [comment]
> Is there any possibility to detect tap on Glabber handle and move each floating state cyclically with each tap?

That's probably possible.

You can change the state of a panel using `FloatingPanelController.move(to:animated:)`. The grabber handle view is too small to detect a tap on it by default, so I recommend to add a tap gesture recognizer to the surface view as you did and then handle the gesture delegate methods to restrict the detectable area and prioritize it with other gesture
